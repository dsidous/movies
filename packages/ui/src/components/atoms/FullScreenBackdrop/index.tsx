import React from 'react';
import Slider from 'react-slick';

import useStyles from './styles';
import 'slick-carousel/slick/slick.css';
import 'slick-carousel/slick/slick-theme.css';

interface Props {
  backdrops: string[];
}

const FullScreenBackdrop: React.FC<Props> = ({ backdrops }) => {
  const settings = {
    arrows: false,
    dots: false,
    fade: true,
    autoplay: true,
    autoplaySpeed: 5000,
    pauseOnHover: false,
    slidesToShow: 1,
    slidesToScroll: 1,
  };
  const classes = useStyles({ height: window.innerHeight });

  if (!backdrops) {
    return null;
  }

  return (
    <Slider {...settings}>
      {backdrops.map(backdrop => (
        <div key={backdrop} className={classes.wrapper}>
          <div
            className={classes.image}
            style={{
              backgroundImage: `url(${backdrop})`,
            }}
          />
        </div>
      ))}
    </Slider>
  );
};

export default FullScreenBackdrop;
