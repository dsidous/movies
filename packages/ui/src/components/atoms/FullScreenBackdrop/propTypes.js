import { array } from 'prop-types';

export const propTypes = {
  backdrops: array.isRequired,
};

export const defaultProps = {};
