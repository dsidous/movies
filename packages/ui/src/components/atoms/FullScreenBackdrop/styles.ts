import { makeStyles } from '@material-ui/core/styles';

export default makeStyles(
  {
    wrapper: {
      float: 'left',
    },
    image: {
      backgroundPosition: 'center center',
      backgroundSize: 'cover',
      backgroundRepeat: 'no-repeat',
      height: props => props.height,
    },
  },
  { name: 'backdropSlide' },
);
