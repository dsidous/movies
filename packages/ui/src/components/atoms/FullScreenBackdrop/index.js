import React from 'react';
import Slider from 'react-slick';

import { propTypes } from './propTypes';
import 'slick-carousel/slick/slick.css';
import 'slick-carousel/slick/slick-theme.css';

const FullScreenBackdrop = ({ backdrops }) => {
  const settings = {
    arrows: false,
    dots: false,
    fade: true,
    autoplay: true,
    autoplaySpeed: 5000,
    adaptiveHeight: true,
    pauseOnHover: false,
    slidesToShow: 1,
    slidesToScroll: 1,
  };

  return (
    <div>
      {backdrops && (
        <Slider {...settings}>
          {backdrops.map(backdrop => (
            <div key={backdrop}>
              <div
                className="fullscreen-backdrop-slide"
                style={{
                  height: window.innerHeight,
                  backgroundImage: `url(${backdrop})`,
                }}
              />
            </div>
          ))}
        </Slider>
      )}
    </div>
  );
};

FullScreenBackdrop.propTypes = propTypes;

export default FullScreenBackdrop;
