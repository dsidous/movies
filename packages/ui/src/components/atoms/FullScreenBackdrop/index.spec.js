import React from 'react';
import { shallow } from 'enzyme';

import FullScreenBackdrop from '.';

const mockProps = {
  backdrops: [
    {
      backdrop: 'path/to/image',
    },
  ],
};

describe('Atoms/FullScreenBackdrop', () => {
  it('should render as expected', () => {
    const wrapper = shallow(<FullScreenBackdrop {...mockProps} />);

    expect(wrapper).toMatchSnapshot();
  });
});
