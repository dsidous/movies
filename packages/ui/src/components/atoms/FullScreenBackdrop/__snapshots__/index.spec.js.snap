// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Atoms/FullScreenBackdrop should render as expected 1`] = `ShallowWrapper {}`;
